File: LeetCodeDaily/28_countUniqueNumbers.py
# 3483. Unique 3-Digit Even Numbers

def solve(digits):
    freq = [0]*10

    for d in digits:
        freq[d] += 1


    cnt = 0
    for i in range(100,1000):

        if i&1:
            continue
        n = i
        a = n%10
        n //= 10
        b = n%10
        n //= 10
        c = n%10

        need = [0]*10
        need[a] += 1
        need[b] += 1
        need[c] += 1

        possible = True

        for d in range(10):
            if need[d] > freq[d] :
                possible = False
                break
        if possible:
            cnt += 1
    return cnt


def main():
    n = int(input())
    digits = list(map(int,input().split()))
    print(solve(digits))

main()
        

File: LeetCodeDaily/28_countUniqueNumbers.cpp
#include<bits/stdc++.h>
using namespace std;
// 3483. Unique 3-Digit Even Numbers
int solve(vector<int>& digits){
    vector<int> freq(10,0);
    for(auto& d : digits){
        freq[d]++;
    }
    int cnt = 0;
    for(int i = 100;i<=999;i++){
        if(i&1) continue;

        int n = i;
        int a = n%10;
        n /= 10;
        int b = n%10;
        n /= 10;
        int c = n%10;

        vector<int> need(10,0);
        need[a]++;
        need[b]++;
        need[c]++;

        bool possible = true;
        for(int d=0;d<=9;d++){
            if(need[d]>freq[d]){
                possible = false;
                break;
            }
        }

        if(possible){
            cnt++;
        }
    }
    return cnt;
    // TC = O(1000)
    // sc = O(1)
}
int main(){
    int n;
    cin >> n;
    vector<int> digits(n);
    for(auto& d : digits){
        cin >> d;
    }

    cout << solve(digits);
    return 0;
}